lib/postgrex/extensions/binary.ex

```diff
@@ -162,10 +162,15 @@ defmodule Postgrex.Extensions.Binary do
       bin = for digit <- digits, into: "", do: <<digit :: uint16>>
       ndigits = div(byte_size(bin), 2)
 
-      [<<ndigits :: int16, weight :: int16, sign :: uint16, scale :: int16>>, bin]
+      <<ndigits :: int16, weight :: int16, sign :: uint16, scale :: int16, bin :: binary>>
     end
   end
 
+  defp pad_to_numeric_base(float) do
+    String.ljust(to_string(float), @numeric_base, ?0)
```

Comment:
That looks great. You can move `String.length(Integer.to_string(@numeric_base))` to a module attribute because the value is constant.